tests/unit/test_charm.py

```diff
@@ -1,7 +1,251 @@
-# Copyright 2023 Canonical
+# Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
 
-class TestCharm:
-    def test_dummy_pass(self):
-        assert True
+import tempfile
+import unittest
+from pathlib import Path
+from unittest.mock import Mock, patch
+
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from scenario import Container, Context, Model, Mount, Relation, State  # type: ignore[import]
+
+from charm import NSSFOperatorCharm
+
+
+@patch("charm.KubernetesServicePatch", new=Mock)
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.ctx = Context(NSSFOperatorCharm)
+
+    def test_given_fiveg_nrf_relation_not_created_when_pebble_ready_then_status_is_blocked(
+        self,
+    ):
+        container = Container(name="nssf", can_connect=True)
```

Comment:
it seems like in all of the tests there is a container and it can connect. Might be worth it putting this line in setUp, and whenever you need to modify the container (e.g. add Mounts), you can do:
```python
container = self.container.replace(mounts=<your mounts>)
state_in = State(containers=[container ...
```

Reply:
Thanks, I updated to `replace`!